### DISCUSSION TITLE
MetaMask and Phantom extensions are installed, but Rainbow isn't recognizing them as installed in my extension.

### DISCUSSION THREAD

**Discussion by azzi907** [rainbow-me/rainbowkit#2226] [Q&A] [answered]
Hi,
I am building a Chrome extension to connect Web3 wallets, and I am using RainbowKit for this purpose. When I run the project on the web, it shows the installed extensions like MetaMask and Phantom as recognized. However, when I load the extension in Chrome, it doesn't display the installed extensions, and there is no option to connect to the wallet through the MetaMask extension. I have to scan from my mobile instead of using the Chrome extension.

**Running as web** 👇 

![image](https://github.com/user-attachments/assets/44a731a0-eed8-4aae-b756-36a6bc938800)

**Loaded as extension in chrome**👇 

![image](https://github.com/user-attachments/assets/9a170b78-8746-4dbc-a766-01696ade5e51)

When I clicked on **metamask** to connect 👇 

![image](https://github.com/user-attachments/assets/201f28c1-7464-4ce2-9645-8aa73e7e61af)

**Comment by DanielSinclair** [comment, marked as answer]
Unfortunately wallet extensions cannot inject their provider into a website hosted by another Chrome extension; this is a Chrome security feature to protect the integrity of extension runtimes.
